.github/workflows/pinot_tests.yml

```diff
@@ -98,8 +101,37 @@ jobs:
           key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
           restore-keys: |
             ${{ runner.os }}-maven-
-      - name: Verify with binary-compatibility-check profile
-        run: mvn verify -DskipTests -Pbinary-compat-check -T1C
+      # Build the PR branch's module to get a new JAR to compare
+      - name: Build SPI on PR branch
+        run: mvn clean package -T1C -pl ${{ matrix.module }} -am -DskipTests -Dproject.build.finalName="${{ matrix.module }}"
+      - name: Checkout master into a subfolder
+        uses: actions/checkout@v4
+        with:
+          ref: master
```

Comment:
Is `master` the latest master or the base of the PR? Ideally we want to checkout the base of the PR